code/modules/events/atmos_leak.dm

```diff
@@ -0,0 +1,90 @@
+//
+// This event causes a gas leak of phoron, sleeping_agent, or carbon_dioxide in a random unoccupied area.
+// One wonders, where did the gas come from? Who knows!  Its SPACE!  But if you want something a touch
+// more "explainable" then check out the canister_leak event instead.
+//
+
+/datum/event/atmos_leak
+	startWhen = 5			// Nobody will actually be in the room, but still give a bit of warning.
+	var/area/target_area	// Chosen target area
+	var/area/target_turf	// Chosen target turf in target_area
+	var/gas_type			// Chosen gas to release
+	// Exclude these types and sub-types from targeting eligibilty
+	var/list/area/excluded = list(
+		/area/shuttle,
+		/area/crew_quarters,
+		/area/holodeck,
+		/area/engineering/engine_room
+	)
+
+// Decide which area will be targeted!
+/datum/event/atmos_leak/setup()
+	var/gas_choices = list("carbon_dioxide", "sleeping_agent") // Annoying
+	if(severity >= EVENT_LEVEL_MODERATE)
+		gas_choices += "phoron" // Dangerous
+	if(severity >= EVENT_LEVEL_MAJOR)
+		gas_choices += "volatile_fuel" // Dangerous and no default atmos setup!
```

Comment:
Yup, I didn't make it up, its a real gas defined in XGM like the other gasses.   It is actually the gas for when welder fuel or stuff is spilled and burned.
It burns like phoron, the difference is that it doesn't show up as named on atmos scan, it shows up as "other" so its more likely techs will assume its okay to use welders and initiate fun(tm), That's why its only an option on the highest severity level.
Please see attached gif for what volatile_fuel does.
![ezgif com-resize](https://cloud.githubusercontent.com/assets/14110581/24002921/58b4ecd0-0a38-11e7-8cde-0b23ddf5ab6b.gif)